third_party/cf-cli-glob/cfignore.go

```diff
@@ -0,0 +1,83 @@
+package glob
+
+import (
+	"path"
+	"strings"
+)
+
+type CfIgnore interface {
+	FileShouldBeIgnored(path string) bool
+}
+
+func NewCfIgnore(text string) CfIgnore {
+	patterns := []ignorePattern{}
+	lines := strings.Split(text, "\n")
+	lines = append(defaultIgnoreLines, lines...)
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+
+		ignore := true
+		if strings.HasPrefix(line, "!") {
+			line = line[1:]
+			ignore = false
+		}
+
+		for _, p := range globsForPattern(path.Clean(line)) {
```

Comment:
I'm not a fan that this library panics if the ignore file is wrong rather than erroring, WDYT about using a different gitignore package like https://github.com/sabhiram/go-gitignore ?